web_ui/ui.go

```diff
@@ -47,6 +50,15 @@ func ConfigureMetrics(engine *gin.Engine) error {
 }
 
 func GetEngine() (*gin.Engine, error) {
+	pc, _, _, ok := runtime.Caller(1)
```

Comment:
I don't think looking at the call stack is the right way to go here.

Does this just mean we should have the `ConfigureMetrics` outside the `GetEngine` call?  That way, the caller can determine the right type of metrics to configure.

Reply:
That should work. I didn't do it just so I won't modify code at multiple places, but I think this approach is more robust than looking at the call stack. Will adopt this. Thanks!